src/bpfilter/cgen/cgen.c

```diff
@@ -190,14 +190,37 @@ int bf_cgen_get_counter(const struct bf_cgen *cgen, uint32_t counter_idx,
 {
     bf_assert(cgen && counter);
 
-    /* There are 1 more counter than number of rules. The last counter is
-     * dedicated to the policy. */
+    /* There are 2 more counter than number of rules. The last counter is
+     * dedicated to the policy. The error counter must be accessed via the
+     * special method below, in order to avoid ambiguity. */
     if (counter_idx > bf_list_size(&cgen->chain->rules))
         return -EINVAL;
 
     return bf_program_get_counter(cgen->program, counter_idx, counter);
 }
 
+int bf_cgen_get_policy_counter(const struct bf_cgen *cgen,
+                              struct bf_counter *counter)
+{
+    bf_assert(cgen && counter);
+
+    // The error slot is always the second last.
+    return bf_program_get_counter(cgen->program,
+                                  bf_list_size(&cgen->chain->rules),
+                                  counter);
+}
+
+int bf_cgen_get_error_counter(const struct bf_cgen *cgen,
+                              struct bf_counter *counter)
+{
+    bf_assert(cgen && counter);
+
+    // The error slot is always the last.
+    return bf_program_get_counter(cgen->program,
+                                  bf_list_size(&cgen->chain->rules)+1,
+                                  counter);
+}
+
```

Comment:
There are some other consequences for this, and I don't *love* abusing the enum as a index also, but I don't have a strong preference. PTAL, and see if this makes sense with the ipt changes to support these semantics (since it will need to change rather than relying on the size of the rules being used to access the policy).

Reply:
It's not ideal, I agree, but I feel the intent is clearer. For `iptables`, the semantics used by `bpfilter` are different, so the bridging between `iptables` and `bpfilter` is expected (same for `nftables`).

Reply:
I've done it in this way.